<?xml version="1.0"?>
<launch>

  <node name="obstacle_detector_node" pkg="lidar_obstacle_detector" type="obstacle_detector_node" output="screen">
    <!-- Input Topic Names -->
    <param name="lidar_points_topic"                  value="/rslidar_points"/>
    <!-- Output Topic Names -->
    <param name="cloud_ground_topic"                  value="obstacle_detector/cloud_ground"/>
    <param name="cloud_clusters_topic"                value="obstacle_detector/cloud_clusters"/>
    <param name="jsk_bboxes_topic"                    value="obstacle_detector/jsk_bboxes"/>
    <param name="autoware_objects_topic"              value="obstacle_detector/objects"/>
    <!-- Parameters -->
    <param name="bbox_target_frame"                   value="velodyne"/>
  </node>

  <node name="danger_publisher_node" pkg="lidar_obstacle_detector" type="danger_publisher_node" output="screen">
      <param name="jsk_bboxes_topic"                    value="obstacle_detector/jsk_bboxes"/>
      <param name="autoware_objects_topic"              value="obstacle_detector/objects"/>
      <param name="reaction_time_allowed"               value="3.0" /> 
      <!--zebra is a smaller width vehicle, try 1.0 total m-->
      <param name="frontal_fov"                         value="0.7" />
  </node>

  <node name="light_controller_node" pkg="lidar_obstacle_detector" type="light_controller_node" output="screen">
      <param name="autoware_danger_topic"               value="/danger_autoware"/>
      <param name="light_on_time"                       value="2.0" /> 
  </node>

<!--
  <node name="simulation_test_node" pkg="lidar_obstacle_detector" type="simulation_test_node" output="screen">
      <param name="autoware_objects_topic"              value="obstacle_detector/objects"/>
      <param name="danger_topic"                        value="/danger_autoware" />
      <param name="num_avgs"                            value="1000" />
  </node>
  -->

  <!-- Dynamic Reconfigure GUI -->
  <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" output="screen" />
  
  <!-- Autoware Object Visualizer (if you have the `lgsvl_utils` pkg, or you can use the official Autoware one) -->
  <!-- <node ns="/obstacle_detector" pkg="lgsvl_utils" type="detected_objects_visualizer" name="autoware_objects_visualizer" output="log" /> -->

  <!-- Rviz -->
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find lidar_obstacle_detector)/rviz/mai_city.rviz" output="log" respawn="true" />
</launch>
